scripts/clear-signed-agreements.py

```diff
@@ -41,13 +47,22 @@ def main(stage, framework_slug, api_token, user):
     )
 
     suppliers = api_client.find_framework_suppliers(framework_slug, agreement_returned=True)['supplierFrameworks']
-    for supplier in suppliers:
+
+    if supplier_ids is not None:
+        supplier_ids = [
+            str(supplier['supplierId']) for supplier in suppliers
```

Comment:
I think the main reason would be to check that they're actually numbers early, before sending them to the API